metricbeat/module/redis/metricset.go

```diff
@@ -0,0 +1,67 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package redis
+
+import (
+	"time"
+
+	rd "github.com/garyburd/redigo/redis"
+
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+// MetricSet for fetching Redis server information and statistics.
+type MetricSet struct {
+	mb.BaseMetricSet
+	pool *rd.Pool
+}
+
+// NewMetricSet creates the base for Redis metricsets
+func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
+	// Unpack additional configuration options.
+	config := struct {
+		IdleTimeout time.Duration `config:"idle_timeout"`
+		Network     string        `config:"network"`
+		MaxConn     int           `config:"maxconn" validate:"min=1"`
+		Password    string        `config:"password"`
+	}{
+		Network:  "tcp",
+		MaxConn:  10,
+		Password: "",
+	}
+	err := base.Module().UnpackConfig(&config)
+	if err != nil {
+		return nil, err
```

Comment:
I'd add more context here. According to https://github.com/elastic/go-ucfg/blob/master/reify.go#L126 in the `go-ucfg` library we use, the error that is going to return doesn't contain any kind of information about the _context_ of the error, just the source of it. It can return messages like `nil value is invalid` but the library itself is unaware of the config files from libbeat so I'd wrap the current error with some specific info like `error reading redis metricbeat config file` or `error reading redis config data`